src/main/java/de/tum/cit/aet/artemis/atlas/service/learningpath/LearningPathService.java

```diff
@@ -243,6 +245,52 @@ private void updateLearningPathProgress(@NotNull LearningPath learningPath) {
         log.debug("Updated LearningPath (id={}) for user (id={})", learningPath.getId(), userId);
     }
 
+    /**
+     * Get the learning path for the current user in the given course.
+     *
+     * @param courseId the id of the course
+     * @return the learning path of the current user
+     */
+    public LearningPathDTO getLearningPathForCurrentUser(long courseId) {
+        final var currentUser = userRepository.getUser();
+        final var learningPath = learningPathRepository.findByCourseIdAndUserIdElseThrow(courseId, currentUser.getId());
+        return LearningPathDTO.of(learningPath);
+    }
+
+    /**
+     * Generate a learning path for the current user in the given course.
+     *
+     * @param courseId the id of the course
+     * @return the generated learning path
+     */
+    public LearningPathDTO generateLearningPathForCurrentUser(long courseId) {
+        final var currentUser = userRepository.getUser();
+        final var course = courseRepository.findWithEagerCompetenciesAndPrerequisitesByIdElseThrow(courseId);
+        if (learningPathRepository.findByCourseIdAndUserId(courseId, currentUser.getId()).isPresent()) {
+            throw new BadRequestException("Learning path already exists.");
+        }
+        final var learningPath = generateLearningPathForUser(course, currentUser);
+        return LearningPathDTO.of(learningPath);
+    }
+
+    /**
+     * Start the learning path for the current user in the given course.
```

Comment:
I don't think there's a "given course" here

```suggestion
     * Start the specified learning path for the current user.
```